### ISSUE TITLE
Install bundled version of npm

### ISSUE THREAD

**Issue by TimonVS** [moonrepo/proto#32]
Hi team, thanks for creating this amazing tool with support for `.nvmrc` and `.node-version`!

I have a feature request: nvm installs the version of npm that comes bundled with the specific version of Node that you're using. Volta allows you to install the bundled npm version with `volta install npm@bundled`. I'd love to see proto provide something similar.

**Comment by milesj** [comment]
Yeah can definitely support this.

**Comment by TimonVS** [comment]
I've been trying this locally. Installing the bundled version of npm works!

Currently, when you have a newer version of npm installed (e.g., you ran `proto install npm`), and you're using `.nvmrc`/`.node-version`, Proto will use the latest version of npm that you have installed. In contrast, nvm and Volta will use the bundled version of npm by default. E.g., Volta:

```console
$ volta install node@16
success: installed and set node@16.19.1 (with npm@8.19.3) as default
$ node -v
v16.19.1
$ npm -v
8.19.3

$ volta install node@16.14.2
success: installed and set node@16.14.2 (with npm@8.5.0) as default
$ node -v
v16.14.2
$ npm -v
8.5.0
```

vs Proto:

```console
$ cat .nvmrc
v16.14.2
$ proto install node
[INFO  proto:install] Installing Node.js with version "latest"
[INFO  proto:install] Node.js has been installed at /Users/timon.vanspronsen/.proto-2/tools/node/19.8.0!
[INFO  proto:install] Installing npm with version "bundled"
[INFO  proto:install] npm has been installed at /Users/timon.vanspronsen/.proto-2/tools/npm/9.5.1!

$ proto install node 16.14.2
[INFO  proto:install] Installing Node.js with version "16.14.2"
[INFO  proto:install] Node.js has been installed at /Users/timon.vanspronsen/.proto-2/tools/node/16.14.2!
[INFO  proto:install] Installing npm with version "bundled"
[INFO  proto:install] npm has been installed at /Users/timon.vanspronsen/.proto-2/tools/npm/8.5.0!
$ node -v
v16.14.2
$ npm -v
9.5.1
```

With Volta you can still install a newer version of npm, from then on it will use this newer version, but you can switch back by running `volta install npm@bundled` (and from then on it will use the bundled version for any version of Node that you install).

```console
$ volta install node@16
success: installed and set node@16.19.1 as default
$ npm -v
8.19.3
$ volta install npm@9
success: installed and set npm@9.6.1 as default
$ volta install node@16
success: installed and set node@16.19.1 as default
$ npm -v
9.6.1
$ volta install npm@bundled
success: set bundled npm (currently 8.5.0) as default
$ volta install node@16.14.2
success: installed and set node@16.14.2 (with npm@8.5.0) as default
$ npm -v
8.5.0
```

**Comment by milesj** [comment]
Thanks for testing it before it's been released!

This should definitely be doable. The only issue is that `engines` or `packageManager` would take precedent if they have an explicit version. Bundled would only really work as the "last case".

**Comment by milesj** [comment]
Released 0.3
